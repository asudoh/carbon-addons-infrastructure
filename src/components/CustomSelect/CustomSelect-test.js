import React from 'react';
import { mount } from 'enzyme';
import { SelectItem } from 'carbon-components-react';
import { CustomSelect, CustomSelectContainer } from './';

describe('CustomSelect', () => {
  it('should render', () => {
    const wrapper = mount(
      <CustomSelectContainer>
        <CustomSelect name="NA North" value="3" groupName="customselect">
          <SelectItem value="TOR02" text="TOR02" />
          <SelectItem value="TOR04" text="TOR04" />
        </CustomSelect>
      </CustomSelectContainer>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
