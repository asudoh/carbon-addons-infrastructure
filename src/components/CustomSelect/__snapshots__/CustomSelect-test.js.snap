// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSelect should render 1`] = `
<CustomSelectContainer>
  <fieldset
    className="infra--custom-select__container"
  >
    <CustomSelect
      groupName="customselect"
      name="NA North"
      value="3"
    >
      <div>
        <input
          hidden={true}
          id="3"
          name="customselect"
          type="radio"
          value="3"
        />
        <label
          className="infra--custom-select"
          htmlFor="3"
        >
          <div
            className="infra--custom-select__icon-container"
          />
          <div
            className="infra--custom-select__select-container"
          >
            <label
              className="infra--custom-select__label"
            >
              NA North
            </label>
            <Select
              className="infra--custom-select__select"
              disabled={false}
              hideLabel={true}
              iconDescription="open list of options"
              inline={true}
              labelText="Select"
            >
              <div
                className="bx--form-item"
              >
                <div
                  className="bx--select bx--select--inline infra--custom-select__select"
                >
                  <label
                    className="bx--label bx--visually-hidden"
                  >
                    Select
                  </label>
                  <select
                    className="bx--select-input"
                    disabled={false}
                  >
                    <SelectItem
                      disabled={false}
                      hidden={false}
                      key=".0"
                      text="TOR02"
                      value="TOR02"
                    >
                      <option
                        className="bx--select-option"
                        disabled={false}
                        hidden={false}
                        value="TOR02"
                      >
                        TOR02
                      </option>
                    </SelectItem>
                    <SelectItem
                      disabled={false}
                      hidden={false}
                      key=".1"
                      text="TOR04"
                      value="TOR04"
                    >
                      <option
                        className="bx--select-option"
                        disabled={false}
                        hidden={false}
                        value="TOR04"
                      >
                        TOR04
                      </option>
                    </SelectItem>
                  </select>
                  <Icon
                    className="bx--select__arrow"
                    description="open list of options"
                    fillRule="evenodd"
                    name="caret--down"
                    role="img"
                  >
                    <svg
                      alt="open list of options"
                      aria-label="open list of options"
                      className="bx--select__arrow"
                      fillRule="evenodd"
                      height="5"
                      name="caret--down"
                      role="img"
                      viewBox="0 0 10 5"
                      width="10"
                    >
                      <title>
                        open list of options
                      </title>
                      <path
                        d="M10 0L5 5 0 0z"
                        key="key0"
                      />
                    </svg>
                  </Icon>
                </div>
              </div>
            </Select>
          </div>
        </label>
      </div>
    </CustomSelect>
  </fieldset>
</CustomSelectContainer>
`;
